code/game/objects/items/weapons/inducer.dm

```diff
@@ -0,0 +1,218 @@
+
+/obj/item/weapon/inducer
+	name = "inducer"
+	desc = "A tool for inductively charging internal power cells."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "inducer-engi"
+	item_state = "inducer-engi"
+	origin_tech = "engineering=4;magnets=4;powerstorage=4"
+	force = 7
+	var/powertransfer = 1000
+	var/opened = FALSE
+	var/cell_type = /obj/item/weapon/stock_parts/cell/high
+	var/obj/item/weapon/stock_parts/cell/cell
+
+/obj/item/weapon/inducer/Initialize()
+	. = ..()
+	src.cell = new cell_type
+
+/obj/item/weapon/inducer/proc/induce(obj/item/weapon/stock_parts/cell/target)
+	var/totransfer = min(cell.charge,powertransfer)
+	var/transferred = target.give(totransfer)
+	cell.use(transferred)
+	cell.updateicon()
+	target.updateicon()
+
+/obj/item/weapon/inducer/proc/get_cell(atom/target)
+	if(istype(target, /obj/machinery/power/apc))
```

Comment:
Eww.
```
var/success = hasvar(target, "cell")
if(success)
    return target:cell // : is ok as we've checked above.
```

This returns the ```cell``` variable if they have one, null otherwise (```return null``` isn't necessary)